compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -176,7 +176,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             self.demand_eqtype(span, inferred_kind.to_ty(self.tcx), closure_kind_ty);
 
             // If we have an origin, store it.
-            if let Some(origin) = delegate.current_origin {
+            if let Some(origin) = delegate.current_origin.clone() {
+                let origin = if self.tcx.features().capture_disjoint_fields {
+                    origin
+                } else {
+                    (origin.0, Place { projections: vec![], ..origin.1 })
```

Comment:
Can you leave a comment here saying.

// FIXME(project-rfc-2229#26): Once #80092, we should restrict the precision of origin as well. Otherwise, this will cause issues when project-rfc-2229#26 is fixed as we might see Index projections in the origin, which we can't print because we don't store enough information.